File: core/model/src/commonMain/kotlin/io/github/droidkaigi/confsched/model/Contributor.kt
package io.github.droidkaigi.confsched.model

import kotlinx.collections.immutable.PersistentList
import kotlinx.collections.immutable.toPersistentList

public data class Contributor(
    val id: Int,
    val username: String,
    val profileUrl: String?,
    val iconUrl: String,
) {
    public companion object
}

public fun Contributor.Companion.fakes(): PersistentList<Contributor> = (1..20)
    .map {
        Contributor(
            id = it,
            username = "username $it",
            profileUrl = "https://developer.android.com/",
            iconUrl = "https://placehold.jp/150x150.png",
        )
    }
    .toPersistentList()

File: app-ios/Sources/Model/Entity/Contributor.swift
import Foundation

public struct Contributor: Equatable, Sendable, Identifiable {
    public var id: Int
    public let userName: String
    public let profileUrl: URL?
    public let iconUrl: URL
    
    public init(id: Int, userName: String, profileUrl: URL?, iconUrl: URL) {
        self.id = id
        self.userName = userName
        self.profileUrl = profileUrl
        self.iconUrl = iconUrl
    }
}
